pdns/syncres.hh

```diff
@@ -1012,8 +1012,10 @@ public:
   enum stateenum {BYTE0, BYTE1, GETQUESTION, DONE} state{BYTE0};
   uint16_t qlen{0};
   uint16_t bytesread{0};
-
+  std::atomic<int> d_requestsInFlight;
```

Comment:
Does it really need to be atomic? I don't think we have any concurrency issue?

Reply:
I wasn't sure if tcpconnection was read from multiple threads. Will double check.

Reply:
Indeed, TCPConnections are only shared by mthreads